vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -991,4 +1006,60 @@ public Style getStyle() {
                 "Dialog does not support adding styles to overlay");
     }
 
+    static class OverlayClassListProxy extends AbstractSet<String>
+            implements ClassList {
+        private final HasStyle hasStyle;
+        private final ClassList classList;
+
+        public OverlayClassListProxy(HasStyle hasStyle) {
+            this.hasStyle = hasStyle;
+            this.classList = hasStyle.getElement().getClassList();
+        }
+
+        private void updateOverlayClass() {
+            hasStyle.getElement().setProperty("overlayClass",
+                    hasStyle.getClassName());
+        }
+
+        @Override
+        public Iterator<String> iterator() {
+            return new IteratorProxy(classList.iterator());
+        }
+
+        @Override
+        public int size() {
+            return classList.size();
+        }
+
+        @Override
+        public boolean add(String s) {
```

Comment:
Maybe I missed something, but don't we need `remove` and `clear` methods here still to update the overlay class? Or how the overlay class is updated if I call `dialog.getClassNames().remove(myStyleClass)`?

Reply:
This works due to how `AbstractSet` and `AbstractCollection` are implemented - they always use the iterator and its `remove` method for any removal operation. So just updating the overlay class name in the iterator proxy's `remove` method covers all these cases.

Reply:
Thanks, now I see.